tests/admin_views/tests.py

```diff
@@ -5132,6 +5135,44 @@ def test_readonly_foreignkey_links(self):
         # Related ForeignKey object not registered in admin.
         self.assertContains(response, '<div class="readonly">Chapter 1</div>', html=True)
 
+    def test_readonly_foreignkey_links_custom_adminsite(self):
+        """
+        Alternate admin site generates proper links for readonly fields
+        """
+        chapter = Chapter.objects.create(
+            title='Chapter 1',
+            content='content',
+            book=Book.objects.create(name='Book 1'),
+        )
+        language = Language.objects.create(iso='_40', name='Test')
+        obj = ReadOnlyRelatedField.objects.create(
+            chapter=chapter,
+            language=language,
+            user=self.superuser,
+        )
+
+        request = RequestFactory().get(
+            reverse(
+                'namespaced_admin:admin_views_readonlyrelatedfield_change',
+                args=(obj.pk,)
+            )
+        )
+
+        # Get an AdminReadonlyField for the alternate admin site
+        related_admin = site2._registry[ReadOnlyRelatedField]
+        admin_form = related_admin.get_form(request, obj=obj)()
+        readonly_field = AdminReadonlyField(
+            admin_form, 'chapter', True, model_admin=related_admin
+        )
+
+        # Verify that a foreign key to the same app renders correctly
+        field_url = readonly_field.get_admin_url(Language.iso.field, language)
+        self.assertIn('/admin5/', field_url)
+
+        # Verify that a foriegn key to a different app renders correctly
+        field_url = readonly_field.get_admin_url(User.id.field, self.superuser)
+        self.assertIn('/admin5/auth/user/', field_url)
```

Comment:
We could reuse `test_readonly_foreignkey_links()`, e.g.
```diff
diff --git a/tests/admin_views/tests.py b/tests/admin_views/tests.py
index a8524379f8..7e0a3a3415 100644
--- a/tests/admin_views/tests.py
+++ b/tests/admin_views/tests.py
@@ -5093,7 +5093,7 @@ class ReadonlyTest(AdminFieldExtractionMixin, TestCase):
         response = self.client.get(reverse('admin:admin_views_choice_change', args=(choice.pk,)))
         self.assertContains(response, '<div class="readonly">No opinion</div>', html=True)
 
-    def test_readonly_foreignkey_links(self):
+    def _test_readonly_foreignkey_links(self, admin_site):
         """
         ForeignKey readonly fields render as links if the target model is
         registered in admin.
@@ -5110,10 +5110,10 @@ class ReadonlyTest(AdminFieldExtractionMixin, TestCase):
             user=self.superuser,
         )
         response = self.client.get(
-            reverse('admin:admin_views_readonlyrelatedfield_change', args=(obj.pk,)),
+            reverse(f'{admin_site}:admin_views_readonlyrelatedfield_change', args=(obj.pk,)),
         )
         # Related ForeignKey object registered in admin.
-        user_url = reverse('admin:auth_user_change', args=(self.superuser.pk,))
+        user_url = reverse(f'{admin_site}:auth_user_change', args=(self.superuser.pk,))
         self.assertContains(
             response,
             '<div class="readonly"><a href="%s">super</a></div>' % user_url,
@@ -5121,7 +5121,7 @@ class ReadonlyTest(AdminFieldExtractionMixin, TestCase):
         )
         # Related ForeignKey with the string primary key registered in admin.
         language_url = reverse(
-            'admin:admin_views_language_change',
+            f'{admin_site}:admin_views_language_change',
             args=(quote(language.pk),),
         )
         self.assertContains(
@@ -5132,6 +5132,12 @@ class ReadonlyTest(AdminFieldExtractionMixin, TestCase):
         # Related ForeignKey object not registered in admin.
         self.assertContains(response, '<div class="readonly">Chapter 1</div>', html=True)
 
+    def test_readonly_foreignkey_links_default_admin_site(self):
+        self._test_readonly_foreignkey_links('admin')
+
+    def test_readonly_foreignkey_links_custom_admin_site(self):
+        self._test_readonly_foreignkey_links('namespaced_admin')
+
     def test_readonly_manytomany_backwards_ref(self):
         """
         Regression test for #16433 - backwards references for related **objects**
```

Reply:
I have a question about this. Yes, you're absolutely correct that it could be done this way. However, it appears to me that the scope of the existing test is far more broad than it needs to be. As far as I can tell, it's effectively testing the reverse function for the name in addition to just testing the get_admin_url function.
 So, based on this comment that I received from Carlton at https://forum.djangoproject.com/t/input-on-ticket-33077/9589/2:
>  I’d say first option is to add a test next to test_readonly_foreignkey_links (or adjust that, maybe) to check the custom site case — but yes that test is quite high-level. If you want to write a narrower unit-test style case, that’s super. (Improving coverage, or specificity is allowed and welcomed.)
I interpreted this as saying that the revised test structure would be considered appropriate. (I acknowledge that I may have read too much into this. I'm fine either way.)

Please advise.

Reply:
`get_admin_url()` is a small internal hook, so I don't think we need separate tests for it. We also try to keep patches intended for backporting as small as possible.

If you want to add extra tests, I would do this in a separate commit/PR that won't be backported (\cc @carltongibson).

Reply:
Yes. The patch for 3.2 is (ideally) as small as possible — then we can improve the tests on `main`. 

I do agree that the existing test is a little broad. (It's integration vs unit tests. Both have their place.)